mm2src/coins/eth/wallet_connect.rs

```diff
@@ -0,0 +1,286 @@
+/// https://docs.reown.com/advanced/multichain/rpc-reference/ethereum-rpc
+use crate::common::Future01CompatExt;
+use crate::Eip1559Ops;
+use crate::{BytesJson, MarketCoinOps, TransactionErr};
+
+use common::log::info;
+use common::u256_to_hex;
+use derive_more::Display;
+use enum_derives::EnumFromStringify;
+use ethcore_transaction::{Action, SignedTransaction};
+use ethereum_types::H256;
+use ethereum_types::{Address, Public, H160, H520, U256};
+use ethkey::{public_to_address, Message, Signature};
+use kdf_walletconnect::WalletConnectOps;
+use kdf_walletconnect::{chain::{WcChainId, WcRequestMethods},
+                        error::WalletConnectError,
+                        WalletConnectCtx};
+use mm2_err_handle::prelude::*;
+use secp256k1::PublicKey;
+use secp256k1::{recovery::{RecoverableSignature, RecoveryId},
+                Secp256k1};
+use std::iter::FromIterator;
+use std::str::FromStr;
+use web3::signing::hash_message;
+
+use super::{EthCoin, EthPrivKeyPolicy};
+
+// Wait for 60 seconds for the transaction to appear on the RPC node.
+const WAIT_RPC_TIMEOUT_SECS: u64 = 60;
+
+#[derive(Display, Debug, EnumFromStringify)]
+pub enum EthWalletConnectError {
+    UnsupportedChainId(WcChainId),
+    InvalidSignature(String),
+    AccoountMisMatch(String),
+    #[from_stringify("rlp::DecoderError", "hex::FromHexError")]
+    TxDecodingFailed(String),
+    #[from_stringify("ethkey::Error")]
+    InternalError(String),
+    InvalidTxData(String),
+    SessionError(String),
+    WalletConnectError(WalletConnectError),
+}
+
+impl From<WalletConnectError> for EthWalletConnectError {
+    fn from(value: WalletConnectError) -> Self { Self::WalletConnectError(value) }
+}
+
+/// Eth Params required for constructing WalletConnect transaction.
+pub struct WcEthTxParams<'a> {
+    pub(crate) gas: U256,
+    pub(crate) nonce: U256,
+    pub(crate) data: &'a [u8],
+    pub(crate) my_address: H160,
+    pub(crate) action: Action,
+    pub(crate) value: U256,
+    pub(crate) gas_price: Option<U256>,
+    pub(crate) chain_id: u64,
+    pub(crate) max_fee_per_gas: Option<U256>,
+    pub(crate) max_priority_fee_per_gas: Option<U256>,
+}
+
+impl<'a> WcEthTxParams<'a> {
+    /// Construct WalletConnect transaction json from from `WcEthTxParams`
+    fn prepare_wc_tx_format(&self) -> MmResult<serde_json::Value, EthWalletConnectError> {
+        let mut tx_object = serde_json::Map::from_iter([
+            ("chainId".to_string(), json!(self.chain_id)),
+            ("nonce".to_string(), json!(u256_to_hex(self.nonce))),
+            ("from".to_string(), json!(format!("0x{:x}", self.my_address))),
```

Comment:
I believe we have a display address function that can be used in these situations

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2223/commits/3db29387107661f5402f14f0c48d441b20a62e26